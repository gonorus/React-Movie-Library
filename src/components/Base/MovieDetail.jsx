import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

import Modal from './Modal';
import Console from '../../helpers/Console';

const MovieDetail = ({ url }) => {
  useEffect(() => {
    Console.Log('mounting', url);
  }, []);

  return (
    <Modal>
      <div className='movie-details'>
        {url}
      </div>
    </Modal>
  );
};
MovieDetail.propTypes = {
  url: PropTypes.string.isRequired
};
export default MovieDetail;
