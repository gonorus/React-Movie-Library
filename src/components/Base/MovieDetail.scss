.movie-details-container {
  max-width: 300px;
  max-height: 500px;
  padding: 16px;
  display: grid;
  row-gap: 8px;
  overflow: auto;

  img {
    max-height: 300px;
    max-width: 200px;
  }

  p {
    margin: 0;
  }

  .showbox {
    position: relative;
  }

  .movie-info {
    pre {
      white-space: pre-wrap;
    }
    
    p {
      text-align: justify;
    }
  }

  &::-webkit-scrollbar {
    height: 6px;
    width: 6px;
    background: #FFF;
  }

  &::-webkit-scrollbar-thumb {
    background: #393812;
    -webkit-border-radius: 1em;
  }
}