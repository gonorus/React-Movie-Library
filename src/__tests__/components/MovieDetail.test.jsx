import React from 'react';
import renderer from 'react-test-renderer';
import MovieDetail from '../../components/Base/MovieDetail';

test('Movie Detail displayed', () => {
  const component = renderer.create(
    <MovieDetail url='http://localhost/details' />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.unmount();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
