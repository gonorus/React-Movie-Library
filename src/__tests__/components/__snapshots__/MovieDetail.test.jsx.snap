// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Detail displayed 1`] = `
<div
  className="modal-dialog"
>
  <div
    className="movie-details"
  >
    asd
  </div>
</div>
`;

exports[`Movie Detail displayed 2`] = `null`;
